### ISSUE TITLE
Expansion of background subtraction

### ISSUE THREAD

**Issue by noahcabanas** [Quasars/orange-spectroscopy#471]
I wanted to add in some more background subtraction methods into the preprocess widget and wanted to know the best way to implement something like [these](https://github.com/StatguyUser/BaselineRemoval.git). Should it be its own separate interface or should I include it inside of the baseline subtraction option?

**Comment by markotoplak** [comment]
@noahcabanas, currently these would fit best in the baseline subtraction option.

On the other hand, I remember a discussion with @borondics about splitting these and then making option groups. So I do not know about the long term. But these are relatively easy to split/join when there is a well working and tested background code.

**Comment by borondics** [comment]
@noahcabanas, are you trying to add Raman baseline corrections? I would suggest to give them some descriptive name, not too general and then it will be easy to restructure the menu of "Preprocess Spectra" to make a new "Baseline" submenu.

Some people were asking for Asymmetric Least Squares Smoothing for Raman baseline corrections. I found a nice collection of them that we could re-use: https://irfpy.irf.se/projects/ica/_modules/irfpy/ica/baseline.html

If you are interested, Implementing this could help everyone.

**Comment by noahcabanas** [comment]
@borondics Yes I'm trying to add in baseline corrections, sorry for the confusion. The collection you sent has one of the same models as the one I sent so I can definitely use your collection for the code as it seems to have more documentation that covers what is happening and has some other useful methods.  

Regarding the implementation, as I'm still a little confused, would this submenu be within the Baseline correction menu or would you want me to make a new Least Squares smoothing menu within the main drop down selection similar to making a new preprocessor?

**Comment by borondics** [comment]
I think it would be best to make a new general "ALS smoothing" preprocessor that looks similar to the current "Baseline" preprocessor. Then we could implement all the methods in the link above with the parameters available that could be initialized and adjusted as needed based on the selection of the actual subtype of the ALS smoothing.

**Comment by noahcabanas** [comment]
Oh I understand now, I'll get started on it and see what I can do.

**Comment by borondics** [comment]
Great! Let me/us know if you need help with anything!
